### ISSUE TITLE
eslint comma-dangle rule does not apply to function arguments

### ISSUE THREAD

**Issue by furmandev** [typescript-eslint/typescript-eslint#2431]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JS
module.exports = {
  "root": true,
  "env": { "browser": true },
  "parser": "@typescript-eslint/parser",
  parserOptions: {
    project: './tsconfig.json',
    tsconfigRootDir: __dirname,
  },
  "plugins": [
    "@typescript-eslint",
    "spellcheck"
  ],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking"
  ],
  "rules": {
    "comma-dangle": ["error", "always-multiline"],
   ...
```

```TS
constructor(
    private router: Router,
    private account: AccountService // <-- should require comma
  ) { }
```

**Expected Result**

lint should require a comma after the last constructor argument

**Actual Result**

code passes the comma-dangle rule

**Additional Info**
The following is correctly identified as an error:
```TS 
testObject = {
    prop1: "value"
  };
```

But this code also incorrectly passes the rule:
```TS
class Class {
  testFunction(
      arg1: string,
      arg2
    )
}
```

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`      | `^3.10.1` |
| `@typescript-eslint/parser`             | `^3.10.1` |
| `TypeScript`                            | `^3.8.3` |
| `node`                                  | `14.8.0` |
| `npm`                                   | `6.14.5` |

**Comment by bradzacher** [comment]
I'm unable to repro.

![image](https://user-images.githubusercontent.com/7462525/91487921-5db87f00-e863-11ea-9ae8-5702136c7468.png)

---

> - [x] I have updated to the latest version of the packages.

You ticked this box, yet your version of our packages that you listed are clearly not the latest.
Please make sure you follow the template.

**Comment by furmandev** [comment]
My mistake, I had grabbed those version numbers from the wrong package in package.json

I've updated the issue description with the correct version numbers, and double checked I'm on the latest version. The issue still seems to persist, but if it's not reproducible I'll continue to investigate my configuration

**Comment by furmandev** [comment]
Even when I reduce my configuration as simple as I can make it, it seems the issue persists.

.eslintrc.js:
```JS
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "plugins": [
    "@typescript-eslint",
  ],
  "rules": {
    "comma-dangle": ["error", "always-multiline"],
  }
}
```

test.ts:
```TS
abstract class Class {
  constructor(
    private arg1: any,
    private arg2: any
  ) { }
}
```

package.json:
```JSON
{
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^3.10.1",
    "@typescript-eslint/parser": "^3.10.1",
    "codelyzer": "^5.1.2",
    "eslint": "^7.7.0",
    "typescript": "^3.8.3"
  }
}
```

`npx eslint test.ts`
no error output

**Comment by bradzacher** [comment]
It's not well documented at all, but

https://eslint.org/docs/rules/comma-dangle#options-11

> functions should only be enabled when linting ECMAScript 2017 or higher.

If you're using the shorthand config style, then it will always ignore functions unless you set `parserOptions.ecmaVersion` to be a number higher than 8.

**Comment by furmandev** [comment]
I specified `parserOptions.ecmaVersion` as you suggested and it's now working perfectly.

I appreciate it!
